src/Razor/Microsoft.AspNetCore.Razor.Language/src/DirectiveDescriptorBuilderExtensions.cs

```diff
@@ -117,6 +117,28 @@ public static IDirectiveDescriptorBuilder AddAttributeToken(this IDirectiveDescr
             return builder;
         }
 
+        public static IDirectiveDescriptorBuilder AddBooleanToken(this IDirectiveDescriptorBuilder builder)
+        {
+            return AddBooleanToken(builder, name: null, description: null);
+        }
+
+        public static IDirectiveDescriptorBuilder AddBooleanToken(this IDirectiveDescriptorBuilder builder, string name, string description)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.Tokens.Add(
+                DirectiveTokenDescriptor.CreateToken(
+                    DirectiveTokenKind.Boolean,
+                    optional: false,
+                    name: name,
+                    description: description));
+
+            return builder;
+        }
```

Comment:
Would this benefit from having an optional boolean token? So people can just write `@preserveWhitespace` if they want the old behavior.

Reply:
TBH this will be so rarely used I’d prefer to force the more explicit usage. But if you feel differently we can always change this in the future.

Reply:
My concern is not specific to this directive but do you envision more boolean directives in the future that might benefit from this being optional? If so, I think it'll be a little confusing for the users if there are some boolean directive that requires explicit values and some that don't. I don't feel strongly but just something to consider.

Reply:
Totally reasonable points. Perhaps at the time where we first feel it's valuable to add optional boolean support, we can decide whether or not to apply that feature here too, as that won't be a breaking change.